### PR TITLE
New story 6

### PR DESCRIPTION
**Pull request by agallant121**
Complete user story 6

### CODE REVIEW

File: app/controllers/admin/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController
+  before_action :require_admin
+
+  def require_admin
```
Comment (agallant121) [inline app/controllers/admin/base_controller.rb:4]: Add before actions in Admin base controller to limit who can see admin dashboard

File: app/controllers/merchant/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Merchant::BaseController < ApplicationController
+  before_action :require_merchant
+
+  def require_merchant
```
Comment (agallant121) [inline app/controllers/merchant/base_controller.rb:4]: Add before actions in Merchant base controller to limit who can see merchant dashboard

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -18,6 +18,7 @@ def create
   end
 
   def show
+    require_user
```
Comment (agallant121) [inline app/controllers/users_controller.rb:21]: Limits regular visitors from seeing profile page

File: config/routes.rb [hunk 4]
```diff
@@ -42,6 +42,16 @@
   get '/login', to: 'sessions#new'
   post '/login', to: 'sessions#create'
 
-  get '/merchant/dashboard', to: 'dashboards#show'
-  get '/admin/dashboard', to: 'dashboards#show'
+  # get '/merchant/dashboard', to: 'dashboards#show'
+  # get '/admin/dashboard', to: 'dashboards#show'
+
+  namespace :merchant do
+  #only admin users will be able to reach this resource
```
Comment (agallant121) [inline config/routes.rb:49]: Add namespacing for dashboard show pages

File: spec/features/navigation/navigation_spec.rb [hunk 5]
```diff
@@ -96,16 +96,41 @@
...
     expect(page).to_not have_link("Login")
     expect(page).to_not have_link("Register")
     expect(page).to_not have_link("Cart")
   end
+
+  describe "When I try to access any path that begins with /merchant, /admin, /profile" do
+    it "I see a 404 error for /merchant" do
+      visit '/merchant/dashboard'
+      expect(page).to have_content("The page you were looking for doesn't exist (404)")
+    end
```
Comment (agallant121) [inline spec/features/navigation/navigation_spec.rb:112]: Add tests for restrictions on who can visit what pages

**Review by meghanstovall — APPROVED** [review]

**Pull request merged**